src/test/java/org/schemata/parser/SchemaParserTest.java

```diff
@@ -18,12 +24,18 @@
 
   private static Schema userSchema;
 
+  private static DescriptorProtos.FileDescriptorSet descriptorSet;
+
   @BeforeAll
-  static void setUp() {
+  static void setUp() throws IOException {
     List<GeneratedMessageV3> messages = List.of(UserBuilder.User.newBuilder().build());
-    var schemaList = new SchemaParser().parseSchema(messages);
+    var parser = new SchemaParser();
+    var schemaList = parser.parseSchema(messages);
     assertAll("User Schema Sanity Check", () -> assertNotNull(schemaList), () -> assertEquals(1, schemaList.size()));
     userSchema = schemaList.get(0);
+
+    var stream = SchemaParserTest.class.getClassLoader().getResourceAsStream("descriptors/entities.desc");
```

Comment:
This is so cool!!! It reduces the burden from the users to register manually.

Reply:
Yeah totally! I made the interface accept an `InputStream` to allow us more flexibility to read it from a file, STDIN, HTTP request body etc.